File: samples/CSharp/Northwind/OrderDetail.cs
using System;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Order Details")]
   public class OrderDetail {

      [Column(IsPrimaryKey = true)]
      public int OrderID { get; set; }

      [Column(IsPrimaryKey = true)]
      public int ProductID { get; set; }

      [Column]
      public decimal UnitPrice { get; set; }

      [Column]
      public short Quantity { get; set; }

      [Column]
      public float Discount { get; set; }

      [Association(ThisKey = nameof(OrderID))]
      public Order Order { get; set; }

      [Association(ThisKey = nameof(ProductID))]
      public Product Product { get; set; }
   }
}

File: samples/VisualBasic/Northwind/OrderDetail.vb
Imports System
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Order Details")>
   Public Class OrderDetail

      <Column>
      Property Discount As Single

      <Column(IsPrimaryKey:=True)>
      Property OrderID As Integer

      <Column(IsPrimaryKey:=True)>
      Property ProductID As Integer

      <Column>
      Property Quantity As Short

      <Column>
      Property UnitPrice As Decimal

      <Association(ThisKey:=NameOf(OrderID))>
      Property Order As Order

      <Association(ThisKey:=NameOf(ProductID))>
      Property Product As Product

   End Class

End Namespace
